src/vos/vos_iterator.c

```diff
@@ -164,6 +170,13 @@ vos_iter_prepare(vos_iter_type_t type, vos_iter_param_t *param,
 	struct vos_iterator	*iter;
 	int			 rc;
 
+	if (ih == NULL) {
+		D_ERROR("Argument 'ih' is invalid to vos_iter_param\n");
+		return -DER_INVAL;
```

Comment:
"D_ASSERT(ih != NULL)" maybe more clear.

Reply:
I guess it depends on API philosophy.   VOS isn't a user facing API but I'm not a big fan of asserts when someone uses any API incorrectly.

Reply:
Hm, vos_iter_prepare() is not user-faced API, the DAOS internal callers should guarantee that the given parameter is valid.